### PR TITLE
Fix empty predict

### PR DESCRIPTION
**Pull request by yinweisu**
*Issue #, if available:*
https://github.com/awslabs/autogluon/issues/1434

*Description of changes:*
Return empty prediction if given empty dataframe.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tabular/src/autogluon/tabular/learner/abstract_learner.py [hunk 1]
```diff
@@ -154,6 +154,8 @@ def _fit(self, X: DataFrame, X_val: DataFrame = None, scheduler_options=None, hy
         raise NotImplementedError
 
     def predict_proba(self, X: DataFrame, model=None, as_pandas=True, as_multiclass=True, inverse_transform=True):
+        if X.empty:
+            return pd.DataFrame()
```
Comment (Innixma) [inline tabular/src/autogluon/tabular/learner/abstract_learner.py:158]: probably want to return an empty dataframe with the same columns as if it wasn't empty.

File: tabular/tests/unittests/test_tabular.py [hunk 2]
```diff
@@ -239,6 +239,15 @@ def get_benchmark_sets():
     return [toyregres_dataset, binary_dataset, regression_dataset, multi_dataset]
 
 
+def _is_empty_pred(y_pred):
+    if type(y_pred) == np.ndarray:
+        return y_pred.size == 0
+    elif type(y_pred) == pd.DataFrame or type(y_pred) == pd.Series:
+        return y_pred.empty
+    elif type(y_pred) == list:
+        return len(y_pred) == 0
```
Comment (Innixma) [inline tabular/tests/unittests/test_tabular.py:248]: what is the problem with len(y_pred) == 0 for all situations?

**Comment by szha** [comment]
Job PR-1472-4 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1472/4/index.html

File: tabular/src/autogluon/tabular/learner/abstract_learner.py [hunk 3]
```diff
@@ -158,7 +158,9 @@ def predict_proba(self, X: DataFrame, model=None, as_pandas=True, as_multiclass=
             X_index = copy.deepcopy(X.index)
         else:
             X_index = None
-        y_pred_proba = self.load_trainer().predict_proba(self.transform_features(X), model=model)
+        y_pred_proba = []
+        if not X.empty:
+            y_pred_proba = self.load_trainer().predict_proba(self.transform_features(X), model=model)
```
Comment (Innixma) [inline tabular/src/autogluon/tabular/learner/abstract_learner.py:163]: expects np output if not pandas I think, you might want to check.

probably better to do the 4 lines:

```
if X.empty:
    y_pred_proba = np.ndarray([])
else:
    ...
```

Comment (yinweisu) [reply tabular/src/autogluon/tabular/learner/abstract_learner.py:163]: Well, but the method signature says X is a dataframe, that's why I used empty()

**Comment by szha** [comment]
Job PR-1472-6 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1472/6/index.html

**Comment by szha** [comment]
Job PR-1472-7 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1472/7/index.html

**Review by Innixma — APPROVED** [review]
LGTM, nice fix!

**Pull request merged**